pkg/cluster/fetch.go

```diff
@@ -23,20 +23,27 @@ type EksdReleaseFetch func(ctx context.Context, name, namespace string) (*eksdv1
 
 type OIDCFetch func(ctx context.Context, name, namespace string) (*v1alpha1.OIDCConfig, error)
 
-func BuildSpecForCluster(ctx context.Context, cluster *v1alpha1.Cluster, bundlesFetch BundlesFetch, eksdReleaseFetch EksdReleaseFetch, gitOpsFetch GitOpsFetch, oidcFetch OIDCFetch) (*Spec, error) {
-	bundles, err := GetBundlesForCluster(ctx, cluster, bundlesFetch)
+type Fetcher interface {
+	BundlesFetch(ctx context.Context, name, namespace string) (*v1alpha1release.Bundles, error)
+	GitOpsFetch(ctx context.Context, name, namespace string) (*v1alpha1.GitOpsConfig, error)
+	EksdReleaseFetch(ctx context.Context, name, namespace string) (*eksdv1alpha1.Release, error)
+	OIDCFetch(ctx context.Context, name, namespace string) (*v1alpha1.OIDCConfig, error)
+}
+
+func BuildSpecForCluster(ctx context.Context, cluster *v1alpha1.Cluster, fetchers Fetcher) (*Spec, error) {
+	bundles, err := GetBundlesForCluster(ctx, cluster, fetchers.BundlesFetch)
 	if err != nil {
 		return nil, err
 	}
-	gitOpsConfig, err := GetGitOpsForCluster(ctx, cluster, gitOpsFetch)
+	gitOpsConfig, err := GetGitOpsForCluster(ctx, cluster, fetchers.GitOpsFetch)
```

Comment:
FWIW, this contributed to my [other comment](https://github.com/aws/eks-anywhere/pull/1593/files#r834413447).

The fact none of the behaviors on the new interface are used in conjunction with each other makes the statement that the behaviors are not cohesive and shouldn't be on the same interface, i.e. this is a code smell.